### PR TITLE
Better detection of a bundler setup

### PR DESCRIPTION
**Pull request by rnelson0**

### CODE REVIEW

**Comment by rnelson0** [comment]
@mckern suggested this is a better way to detect a bundler setup.

File: pre-commit [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 function use_bundle() {
   path_to_bundle=$(command -v bundle)
-  if [ -x "$path_to_bundle" -a -d ".bundle" ] ; then
+  if [[ -f Gemfile ]] && $path_to_bundle check 2>&1 >/dev/null; then
```
Comment (mckern) [inline pre-commit]: Consider not using a bare variable here. `"${path_to_bundle}"`, as opposed to `$path_to_bundle`

**Comment by mckern** [comment]
thrown-under-the-bus-co-worker-throws-teammate-under-the-bus-what-should-you-do-now

Thanks @rnelson0

File: pre-commit (hunk 1 above)
Comment (rnelson0) [inline pre-commit]: Fixed up, thanks.

Comment (roidelapluie) [inline pre-commit]: And use "${path_to_bundle:?bundle not found}" check

File: pre-commit [hunk 2]
```diff
@@ -17,8 +17,13 @@
 
 function use_bundle() {
   path_to_bundle=$(command -v bundle)
-  if [ -x "$path_to_bundle" -a -d ".bundle" ] ; then
-    return 0
+  if [ -x "$path_to_bundle" ] && [[ -f Gemfile ]] ; then
```
Comment (bastelfreak) [inline pre-commit]: we should use [] or [[]] consistently here

Comment (mckern) [inline pre-commit]: Agreed. I broadly prefer `[[]]` when explicitly working with Bash, but if the rest of the script uses `[]`, consistency is valuable.

Comment (mattiasgeniar) [inline pre-commit]: I'll merge this and fix with a quick commit. Thanks for the PR, @rnelson0 !

**Pull request merged**